src/app/api/comments/replies/route.ts

```diff
@@ -0,0 +1,135 @@
+
+    const parsed = createCommentSchema.safeParse(requestBody);
+    console.log("Validation result:", parsed);
+
+    if (!parsed.success) {
+      console.error("Validation error:", parsed.error);
+      return new NextResponse(JSON.stringify({ error: parsed.error }), {
+        status: 400,
+        headers: {
+          "Content-Type": "application/json",
+        },
+      });
+    }
+
+    const replyData = parsed.data;
+
+    const session = await getServerSession();
+    if (session?.user.email !== replyData.userEmail) {
+      return NextResponse.json({ error: "Forbidden" }, { status: 403 });
+    }
+
+    const user = await prisma.user.findUnique({
+      where: { email: replyData.userEmail },
+    });
+
+    if (user === null) {
+      return NextResponse.json({ error: "User not found" }, { status: 404 });
+    }
+
+    const createdReply = await prisma.comment.create({
+      data: {
+        content: replyData.content,
+        quote: replyData.quote,
+        parentCommentId: replyData.parentCommentId,
+        userId: user.id,
+        storyId: replyData.storyId,
+      },
+    });
+
+    return new NextResponse(
+      JSON.stringify({
+        message: "Reply created successfully",
+        commentData: createdReply,
+      }),
+      {
+        status: 200,
+        headers: {
+          "Content-Type": "application/json",
+        },
+      },
+    );
+  } catch (error) {
+    console.log(error);
+    console.log("Received request:", req.method, req.url);
+    console.error("Error processing request:", error);
+    return new NextResponse(
+      JSON.stringify({ error: "Internal Server Error!" }),
+      {
+        status: 500,
+        headers: {
+          "Content-Type": "application/json",
+        },
+      },
+    );
+  }
+}
+
+// to get all replies to comments made by the user(userEmail)
+export async function GET(req: NextRequest) {
+  try {
+    const url = new URL(req.url);
+    const userEmail = url.searchParams.get("userEmail");
+
+    if (!userEmail) {
+      return new NextResponse(JSON.stringify({ error: "userId is required" }), {
+        status: 400,
+        headers: { "Content-Type": "application/json" },
+      });
+    }
+
+    // const session = await getServerSession();
+    // if (session?.user.email !== userEmail) {
+    //   return NextResponse.json({ error: "Forbidden" }, { status: 403 });
+    // }
+
+    const user = await prisma.user.findUnique({
+      where: { email: userEmail },
+    });
+
+    if (user === null) {
+      return NextResponse.json({ error: "User not found" }, { status: 404 });
+    }
+
+    const userComments = await prisma.comment.findMany({
+      where: { userId: user.id },
+      select: {
+        id: true,
+      },
+    });
+
+    const userCommentIds = userComments.map((comment) => comment.id);
+
+    const replies = await prisma.comment.findMany({
+      where: {
+        parentCommentId: {
+          in: userCommentIds,
+        },
+      },
+    });
```

Comment:
I added a comment about sorting by related documents elsewhere.  Do you think it's worth it to change to sorting by related docs in one search instead of two? I don't really feel like this is that big of a deal, though, so I'd like to know what you think